scribeauth/scribeauth.py

```diff
@@ -39,20 +51,48 @@ class TooManyRequestsException(Exception):
     """
     pass
 
+class MissingIdException(Exception):
+    pass
+
+class UnknownException(Exception):
+    pass
+
+def is_complete_credentials(cred: Credentials):
+    return cred and cred.access_key_id and cred.secret_key and cred.session_token
+
 class ScribeAuth:
-    def __init__(self, client_id: str):
+    def __init__(self, param: Union[Unpack[Federation], str]):
         """Construct an authorisation client.
 
         Args
         ----
+        Union[Unpack[Federation], str]
+        A parameter that can either be an instance of Federation or a string.
+
+        Federation:
+        
+        client_id -- The client ID of the application provided by Scribe.
+        
+        user_pool_id -- The user pool ID provided by Scribe.
+        
+        identity_pool_id -- The identity pool ID provided by Scribe.
+        ----
+        .. deprecated:: Use the constructor with an object instead.
```

Comment:
Does this make sense? There's no object in Python, and also with the union instead of the override there's nothing deprecated